ToolBox/IEX/IEXEventSourceCollection.cs

```diff
@@ -0,0 +1,208 @@
+                    packagedSymbolsList.Add(remainingSymbols.ToArray());
+                    break;
+                }
+            }
+            while (remainingSymbols.Any());
+
+            // Create new client for every package (make sure that we do not exceed the rate-gate-limit while creating)
+            packagedSymbolsList.DoForEach(package =>
+            {
+                _rateGate.WaitToProceed();
+
+                Log.Debug($"IEXEventSourceCollection.CreateNewSubscription(): Creating new subscription for: {string.Join(",", package)}");
+                var client = CreateNewSubscription(package);
+
+                // Add to the dictionary
+                ClientSymbolsDictionary.TryAdd(client, package);
+            });
+
+            Counter.Signal();
+            if (!Counter.Wait(TimeoutToUpdate))
+            {
+                throw new Exception("IEXEventSourceCollection.UpdateSubscription(): Could not update subscription within a timeout");
+            }
+
+            clientsToRemove.DoForEach(RemoveOldClient);
+
+            // Reset counter
+            Counter.Reset(1);
+
+            IsConnected = true;
+        }
+
+        protected virtual EventSource CreateNewSubscription(string[] symbols)
+        {
+            var client = CreateNewClient(symbols);
+
+            // Set up the handlers
+            client.Opened += (sender, args) =>
+            {
+                Counter.Signal();   // Decrement
+                Log.Debug($"ClientOnOpened(): Counter count after decrement: {Counter.CurrentCount}");
+            };
+
+            client.MessageReceived += _messageAction;
+
+            client.Error += (sender, args) =>
+            {
+                var exception = args.Exception;
+                Log.Debug($"ClientOnError(): EventSource Error Occurred. Details: {exception.Message} " +
+                          $"ErrorType: {exception.GetType().FullName}");
+            };
+
+            client.Closed += (sender, args) =>
+            {
+                Log.Debug("ClientOnClosed(): Closing a client");
+            };
+
+            // Client start call will block until Stop() is called (!) - runs continuously in a background
+            Task.Run(async () => await client.StartAsync().ConfigureAwait(false));
+
+            return client;
+        }
+
+        protected EventSource CreateNewClient(string[] symbols)
+        {
+            var url = BuildUrlString(symbols);
+            var client = new EventSource(LaunchDarkly.EventSource.Configuration.Builder(new Uri(url)).Build());
+            return client;
+        }
+
+        protected virtual void RemoveOldClient(EventSource client)
+        {
+            Log.Debug($"IEXEventSourceCollection.UpdateSubscription(): Remove subscription for: {string.Join(",", ClientSymbolsDictionary[client])}");
+
+            string[] stub;
+            ClientSymbolsDictionary.TryRemove(client, out stub);
+
+            client.Dispose();
```

Comment:
Nice 👍 
We have a `DisposeSafely` extension method which is usefull because it handles any explosion internally, suggest we can use it instead